src/app/task-details/task-details.component.ts

```diff
@@ -27,27 +30,48 @@ export class TaskDetailsComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    const taskId = +this.route.snapshot.paramMap.get('id')!;
-    this.task = this.taskService.getTasks()[taskId];
-    if (this.task){
-      this.taskForm.setValue({
-        title: this.task.title,
-        description: this.task.description,
-        type: this.task.type,
-        status: this.task.status
-      });
+    const taskId = this.route.snapshot.paramMap.get('id');
```

Comment:
there's a high chance this value might be undefined/null

Reply:
therefore under this you won't be able to fetch the task 100% of the time 
I would suggest using rxjs' operators to chain multiple operations together, first get the id, then switchMap to this.getTask(id)